src/main/capi/prepared-c.cpp

```diff
@@ -31,8 +31,10 @@ idx_t duckdb_extract_statements(duckdb_connection connection, const char *query,
 	Connection *conn = reinterpret_cast<Connection *>(connection);
 	try {
 		wrapper->statements = conn->ExtractStatements(query);
-	} catch (const duckdb::ParserException &e) {
```

Comment:
I suppose this is the actual fix? My rework is fixing this as well since we're moving all of the catches to `std::exception` everywhere. I'll add the test case to my rework.

Reply:
Yes, this is the fix.
> I'll add the test case to my rework.

Perfect! So we'll only catch `std::exception` in the C API from now on?

Also, should I then close this PR? What are the plans for the `FIXME`s, i.e., cases where we only return `DuckDBError` to the user, and omit the actual exception message?

Reply:
Yes, we'll only need to catch `std::exception` after my rework (in the C API and elsewhere).

Let's close this PR to avoid merge conflicts - if required you can add the FIXME's later. The problem is that we can't always easily return the actual error message (and often it is not interesting either, e.g. a lot of the catches are there just to catch memory allocation failures so they basically never happen in practice).